src/effects/VST3/VST3Effect.cpp

```diff
@@ -671,7 +735,14 @@ size_t VST3Effect::ProcessBlock(EffectSettings &,
 
 bool VST3Effect::RealtimeInitialize(EffectSettings &settings, double sampleRate)
 {
-   mSetup.sampleRate = sampleRate;
+   if(mSetup.sampleRate != sampleRate)
+   {
+      auto setup = mSetup;
+      setup.sampleRate = sampleRate;
+      if(!SetupProcessing(*mEffectComponent.get(), setup))
```

Comment:
In this place, you are not careful, as with other calls to `SetupProcessing` to avoid side effects on mSetup in case of failure.  Should that change?

There are also two calls in constructors of VST3Effect but I think I understand that it is not necessary there.